config/crd/patches/more_validations_in_bindingpolicies.yaml

```diff
@@ -0,0 +1,12 @@
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/clusterSelectors/items/properties/matchLabels/additionalProperties/pattern
+  value: "^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$"
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/clusterSelectors/items/properties/matchExpressions/items/properties/key/pattern
+  value: ^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/clusterSelectors/items/properties/matchExpressions/items/properties/operator/pattern
+  value: "^(In|NotIn|Exists|DoesNotExist)$"
```

Comment:
This could be done more simply and directly with https://json-schema.org/understanding-json-schema/reference/enum instead. Not a blocker for this PR.